### DISCUSSION TITLE
Retraining and Directories sorting

### DISCUSSION THREAD

**Discussion by YezCohen** [ultralytics/yolov5#9856] [Q&A]
Hi there to all, 

I have two questions that I would be happy to get help with and hear your thoughts on.
I am new to creating object detection models with yolov5, so in case I'm not explaining myself right or in the correct manner, please correct me.

I have 4 models that I want to train. Three will know to detect only one class in every image, and the fourth will know to detect multiple classes in every image (Not necessarily the same classes as the other 3).

I have currently a dataset of about 10,000 images for each one of those models that include the classes that I want to detect, and that dataset will continuously grow (I will have about 2000 new images for each model every couple of weeks.

So my questions are as follows:
**1:** Directories and file organizing - what do you think would be the best approach how to organizing all those images from the start?
**2:** Retraining - As I said I will have new data to add and retrain the model every few weeks, I tried finding previously posted topics about it, and in posts online if there is a way to add the weights of the trained model with only the new images and retrain it or I will have to add the new images to the old dataset and retrain it from the beginning on the entire set. I found contradicting answers (or I misunderstood, so correct me if I'm wrong please) [here](https://stackoverflow.com/questions/68993575/yolov5-custom-retraining) and [here](https://towardsdatascience.com/the-practical-guide-for-object-detection-with-yolov5-algorithm-74c04aac4843) (under "Hot start from pretrained model" paragraph).

In case I can retrain with only the new images, It would be awesome if you could please refer me to where I can learn how to do so.
Otherwise - is there a common way of going about it of "filtering" out old images and using new ones, at some point so I won't find myself retraining the model from scratch on tens of thousands of images every few weeks in the future?

Thanks in advance :)

**Comment by glenn-jocher** [comment]
@YezCohen you must train a model on all images you want it to learn. If it is not retrained on old images it will forget those, as naturally all parameters are fully dedicated to learning new data.

## Start from Pretrained

If you would like to **start training** from a fully trained model, use the `--weights` argument, not the `--resume` argument:
```bash
python train.py --weights path/to/best.pt  # start from pretrained model
```

Good luck 🍀 and let us know if you have any other questions!

**Reply by YezCohen** [reply]
> @YezCohen you must train a model on all images you want it to learn. If it is not retrained on old images it will forget those, as naturally all parameters are fully dedicated to learning new data.
> ## Start from Pretrained
> 
> If you would like to **start training** from a fully trained model, use the `--weights` argument, not the `--resume` argument:
> 
> ```shell
> python train.py --weights path/to/best.pt  # start from pretrained model
> ```
> 
> Good luck four_leaf_clover and let us know if you have any other questions!

Thank you for the quick response Glenn!
I understand now, will train on the entire dataset including the "old" images.
Do you have any thoughts or knowledge regarding my last question? 

> Otherwise - is there a common way of going about it of "filtering" out old images and using new ones, at some point so I won't find myself retraining the model from scratch on tens of thousands of images every few weeks in the future?

meaning - I can retrain the model every few weeks with the best weights saving time. But is there a storage-friendly way to somehow keep training and improving the model but not keeping ALL of the old images?

Thanks!

**Reply by glenn-jocher** [reply]
@YezCohen I think in environments where there is an excess of data you need to determine a way to prioritize which data to train and which to exclude, probably based on class metrics, i.e. include more images of underperforming classes.
